Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/loader/ChanThreadLoader.java

```diff
@@ -603,27 +603,29 @@ public void onErrorResponse(VolleyError error) {
     @Nullable
     private ChanThread loadSavedThreadIfItExists() {
         Loadable loadable = getLoadable();
-        if (loadable != null) {
-            Pin pin = Chan.injector().instance(WatchManager.class).findPinByLoadableId(loadable.id);
-            if (pin == null) {
-                return null;
-            }
+        if (loadable == null) {
+            Logger.d(TAG, "Could not get current loadable, it's null");
+            return null;
+        }
 
-            if (!PinType.hasDownloadFlag(pin.pinType)) {
-                return null;
-            }
+        Pin pin = Chan.injector().instance(WatchManager.class).findPinByLoadableId(loadable.id);
```

Comment:
Uhhh, this PR should be merged only after #305 is merged, Also this line should be changed to watchManager.findPinByLoadableId(loadable.id);

Reply:
Alright.